test_scripts/MobileProjection/Defects/common.lua

```diff
@@ -0,0 +1,191 @@
+m.mobile = actions.mobile
+m.sdl = actions.sdl
+m.run = actions.run
+m.wait = actions.run.wait
+
+--[[ Common Functions ]]
+function m.print(...) utils.cprint(35, ...) end
+
+function m.printTable(...) utils.cprintTable(35, ...) end
+
+function m.timestamp(pEventName, pConType)
+  m.ts[pEventName] = timestamp()
+  local function isExist(pArray, pItem)
+    for _, v in pairs(pArray) do
+      if v == pItem then return true end
+    end
+    return false
+  end
+  local function insert(pTable, pItem)
+    if not isExist(pTable, pItem) then table.insert(pTable, pItem) end
+  end
+  if pConType then
+    insert(m.seq[pConType], pEventName)
+  else
+    for k in pairs(m.seq) do
+      insert(m.seq[k], pEventName)
+    end
+  end
+end
+
+function m.log(...)
+  local str = "[" .. atf_logger.formated_time(true) .. "]"
+  for i, p in pairs({...}) do
+    local delimiter = "\t"
+    if i == 1 then delimiter = " " end
+    str = str .. delimiter .. p
+  end
+  utils.cprint(35, str)
+end
+
+function m.preconditions(pParamValues)
+  actions.preconditions()
+  if pParamValues and type(pParamValues) == "table" then
+    for p, v in pairs(pParamValues) do
+      utils.cprint(35, p, v)
+      m.sdl.setSDLIniParameter(p, v)
+    end
+  end
+end
+
+function m.postconditions()
+  for appId, v in pairs(m.streamingStatus) do
+    if v == true then
+      m.stopStreaming(appId)
+    end
+  end
+  actions.postconditions()
+end
+
+function m.startStreaming(pAppId, pServiceId)
+  m.mobile.getSession(pAppId):StartService(pServiceId)
+  :Do(function()
+      m.mobile.getSession(pAppId):StartStreaming(pServiceId, m.streamFiles[pAppId], 160*1024)
+      m.hmi.getConnection():ExpectNotification("Navigation.OnVideoDataStreaming", { available = true })
+      m.log("App " .. pAppId .." starts streaming ...")
+      m.streamingStatus[pAppId] = true
+    end)
+  m.hmi.getConnection():ExpectRequest("Navigation.StartStream", { appID = m.app.getHMIId(pAppId) })
+  :Do(function(_, data)
+      m.hmi.getConnection():SendResponse(data.id, data.method, "SUCCESS", { })
+    end)
+end
+
+function m.stopStreaming(pAppId)
+  if m.streamingStatus[pAppId] == true then
+    m.mobile.getSession(pAppId):StopStreaming(m.streamFiles[pAppId])
+    m.log("App " .. pAppId .. " stops streaming")
+    m.streamingStatus[pAppId] = false
+  else
+    utils.cprint(33, "Streaming is unable to stop since it's not started")
+  end
+end
+
+function m.registerApp(pAppId)
+  if not pAppId then pAppId = 1 end
+  local session = m.mobile.createSession(pAppId)
+  session:StartService(7)
+  :Do(function()
+      local corId = session:SendRPC("RegisterAppInterface", m.app.getParams(pAppId))
+      m.hmi.getConnection():ExpectNotification("BasicCommunication.OnAppRegistered")
+      :Do(function(_, d)
+          m.app.setHMIId(d.params.application.appID, pAppId)
+        end)
+      session:ExpectResponse(corId, { success = true, resultCode = "SUCCESS" })
+      function session:ExpectEndService(pServiceId)
```

Comment:
Do we need `ExpectEndService` here?

Reply:
@GetmanetsIrina Please see https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/2297#discussion_r358470324